src/jit/flowgraph.cpp

```diff
@@ -1877,6 +1877,48 @@ void Compiler::fgComputeReachabilitySets()
 }
 
 /*****************************************************************************
+ *  Updates BBF_GC_SAFE_POINT flag for each block
+ */
+
+void Compiler::fgUpdateSafepointFlags()
+{
+    if (compTailCallUsed)
+    {
+        for (BasicBlock* block = fgFirstBB; block != nullptr; block = block->bbNext)
+        {
+            block->bbFlags &= ~BBF_GC_SAFE_POINT;
+            for (GenTreeStmt* stmt = block->firstStmt(); stmt != nullptr; stmt = stmt->getNextStmt())
+            {
+                for (GenTree* node = stmt->gtStmtExpr; node != nullptr; node = node->gtNext)
+                {
+                    if (node->IsCall())
+                    {
+                        GenTreeCall* call = node->AsCall();
+                        if (call->IsTailCall())
+                        {
+                            if (call->IsTailCallViaHelper())
+                            {
+                                block->bbFlags |= BBF_GC_SAFE_POINT;
+                            }
+                        }
+                        else
+                        {
+                            block->bbFlags |= BBF_GC_SAFE_POINT;
+                        }
+                    }
+                }
+
+                if ((block->bbFlags & BBF_GC_SAFE_POINT) != 0)
+                {
+                    break;
+                }
+            }
+        }
+        fgComputeReachabilitySets();
```

Comment:
Add a comment explaining what  ` fgComputeReachability()` does to the BBF_GC_SAFE_POINT flags